File: Implementation/Algorithms/cpp/BubbleSort.cpp
#include <bits/stdc++.h>

using namespace std;

void bubbleSort(int arr[], int s) {
    for (int i = 0; i < s; ++i) {
        for (int j = 0; j < s - 1; ++j) {
            if (arr[j] > arr[j + 1]) swap(arr[j], arr[j + 1]);
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int arr[5] = {64, 25, 12, 22, 11};
    int s = sizeof(arr) / sizeof(arr[0]);

    bubbleSort(arr, s);

    for (int i = 0; i < s; ++i) {
        cout << arr[i] << " ";
    }

    cout << "\n";

    return 0;
}

File: Implementation/Algorithms/java/BubbleSort.java
import java.util.*;

public class BubbleSort {
    public static void bubbleSort(int[] arr) {
        for (int i = 0; i < arr.length; ++i) {
            for (int j = 0; j < arr.length - 1; ++j) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;

                    //We also can use
                    //Collections.swap(Arrays.asList(arr), j, j + 1);
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 25, 12, 22, 11};

        bubbleSort(arr);

        for (int i = 0; i < arr.length; ++i)
            System.out.println(arr[i]);
    }
}
